src/com/sun/jna/Platform.java

```diff
@@ -262,8 +262,12 @@ else if ("zarch_64".equals(arch)) {
             arch = "ppc64le";
         }
         // Map arm to armel if the binary is running as softfloat build
-        if("arm".equals(arch) && platform == Platform.LINUX && isSoftFloat()) {
-            arch = "armel";
+        if("arm".equals(arch) && platform == Platform.LINUX ) {
+            if(isSoftFloat()) {
+                arch = "armel";
+            } else if(!is64Bit()) {
+                arch = "armhf";
+            }
```

Comment:
This breaks existing usages. There are currently three variants:

- `arm`: 32 bit, hardfloat
- `armel`: 32bit, softfloat
- `aarch64`: 64bit

If this codepath is hit by 64bit code, I'd like to know where the JVM came from and why the vendor decided not to use `aarch64`  for `os.arch`.

Reply:
I understand that arm implies armhf and that this code is not needed. I'll simplify the change.